client/src/main/java/io/pravega/client/stream/impl/ControllerImpl.java

```diff
@@ -601,21 +602,42 @@ private Boolean handleScaleResponse(Stream stream, ScaleResponse response) {
     }
     
     @Override
-    public CompletableFuture<Set<Segment>> getSuccessors(StreamCut from) {
+    public CompletableFuture<Set<Segment>> getSuccessors(final StreamCut from) {
         Exceptions.checkNotClosed(closed.get(), this);
         Stream stream = from.asImpl().getStream();
         long traceId = LoggerHelpers.traceEnter(log, "getSuccessorsFromCut", stream);
-        HashSet<Segment> unread = new HashSet<>(from.asImpl().getPositions().keySet());
         val currentSegments = getAndHandleExceptions(getCurrentSegments(stream.getScope(), stream.getStreamName()),
-                                                     RuntimeException::new);
-        unread.addAll(computeKnownUnreadSegments(currentSegments, from));
+                                                     RuntimeException::new).getSegments();
+        final Set<Segment> unread = getSegmentsInclusive(from.asImpl(), currentSegments);
+        LoggerHelpers.traceLeave(log, "getSuccessorsFromCut", traceId);
+        return CompletableFuture.completedFuture(unread);
+    }
+
+    @Override
+    public CompletableFuture<Set<Segment>> getSegments(final StreamCut fromStreamCut, final StreamCut toStreamCut) {
+        Exceptions.checkNotClosed(closed.get(), this);
+        Preconditions.checkNotNull(fromStreamCut, "fromStreamCut");
+        Preconditions.checkNotNull(toStreamCut, "toStreamCut");
+        Preconditions.checkArgument(fromStreamCut.asImpl().getStream().equals(toStreamCut.asImpl().getStream()),
+                "Ensure streamCuts for the same stream is passed");
+
+        final Stream stream = fromStreamCut.asImpl().getStream();
+        long traceId = LoggerHelpers.traceEnter(log, "getSuccessorsFromCut", stream);
+        final Set<Segment> segments = getSegmentsInclusive(fromStreamCut, toStreamCut.asImpl().getPositions().keySet());
+        LoggerHelpers.traceLeave(log, "getSuccessorsFromCut", traceId);
+        return CompletableFuture.completedFuture(segments);
+    }
+
+    private Set<Segment> getSegmentsInclusive(StreamCut fromStreamCut, Collection<Segment> currentSegments) {
+        final HashSet<Segment> unread = new HashSet<>(fromStreamCut.asImpl().getPositions().keySet());
```

Comment:
This variable name no longer makes sense.

Reply:
updated the name.